src/main/scala/mesosphere/marathon/state/AppDefinition.scala

```diff
@@ -152,8 +152,11 @@ case class AppDefinition(
         r => r.getName -> (r.getScalar.getValue: JDouble)
       }.toMap
 
+    //The command is executed inside a shell, but the args are not.
+    //Since both attributes set the proto.cmd, we use this flag to
+    //distinguish if the command was set or the args array.
     val commandOption =
-      if (proto.getCmd.hasValue && proto.getCmd.getValue.nonEmpty)
+      if (proto.getCmd.getShell && proto.getCmd.hasValue && proto.getCmd.getValue.nonEmpty)
```

Comment:
It's is not clear what your comment relates to. I guess you should refer to `proto.getCmd.getShell`.